app/controllers/pets_controller.rb

```diff
@@ -47,8 +47,14 @@ def update
 
   def destroy
     @pet = Pet.find(params[:id])
-    @pet.destroy
-    redirect_to "/pets"
+    if !@pet.status
+      flash[:pet_delete_notice] = "Unable to Delete a Pet with an approved application"
+      redirect_to request.referer
+    else
+      favorites.contents.delete(@pet.id)
```

Comment:
This was fun to pry into with you! I am glad we figured out we could just call .delete on our favorites array

Reply:
It really was, it's been a pleasure working with you!!!!